.github/workflows/main.yml

```diff
@@ -0,0 +1,81 @@
+name: Upload to S3
+on:
+    push:
+
+env:
+    AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}  
+    AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}  
+    AWS_SESSION_TOKEN: ${{ secrets.AWS_SESSION_TOKEN }} 
+    AWS_DEFAULT_REGION: us-east-1
+
+jobs:
+    create-resources:
+      runs-on: ubuntu-latest
+        
+      steps:
+        - name: Checkout 
+          uses: actions/checkout@v1
+            
+        - name: Install Terraform
+          uses: hashicorp/setup-terraform@v1
+          
+        - name: Terraform init
+          run: |
+            cd terraform
+            terraform init
+
+        - name: Terraform apply 
+          run: |
+            cd terraform
+            terraform apply -var-file="var.tfvars" -auto-approve
+         
+
+    add-to-s3:
+      needs: create-resources
+      runs-on: ubuntu-latest
+      steps:
+        - name: Checkout code
+          uses: actions/checkout@v3
+          
+        - name: Configure aws credentials
+          uses: aws-actions/configure-aws-credentials@v1
+          with:
+            aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}  
```

Comment:
why use this when you have define environment

Reply:
passes credentials using environment